### DISCUSSION TITLE
🎉【第1回】2024年08月18日

### DISCUSSION THREAD

**Discussion by kagomen** [lgtm-factory/lgtm-factory#45] [Meeting report]
### 目次

目次はトピックを立てた人が書き加えてください
- [Week1関連の確認](https://github.com/lgtm-factory/lgtm-factory/discussions/45#discussioncomment-10301817)
    - 結論：コードレビューに関しては、絶対満たすべきポイントを押さえていれば、各自の基準でOK！そこまで厳密ではない！
- [Week2関連の確認](https://github.com/lgtm-factory/lgtm-factory/discussions/45#discussioncomment-10369671)
    - 結論：FigJam内の内容を目安に、ワイヤーフレームやイシューを作成し、少しずつ具体的なイメージを掴んでいく！
- [デザインカンプについて](https://github.com/lgtm-factory/lgtm-factory/discussions/45#discussioncomment-10364537)
    - 結論：一旦ワイヤーフレームを作成し、それを元に細かい確認や期限決めを行う！
- [これからの流れ](https://github.com/lgtm-factory/lgtm-factory/discussions/45#discussioncomment-10364564)
    - 結論：
        - 焦点は主に２つ → デザインカンプの作成と、機能の実装。
        - 加えて、コツコツLGTMのベースデザインを溜めていく！
    

### 次の一週間でやること
ざっくり
- デザインカンプの作成
- 機能の実装
-コツコツLGTMのベースデザインを溜めていく！

### トピック用フォーマット

以下をコピーしてトピックを立ててください
```
## トピック：
<!-- 概要 -->

## 結論

## 関連Issues, Discussions
<!-- あれば書いてください -->
```

**Comment by kagomen** [comment]
## トピック：Week1関連の確認
- Discord、Issues、Discussionsの使い分けを確認
- イシューベースの開発の流れの確認
- ブランチ戦略
- コミットメッセージ
- PRやコードレビューの手順
- 

## 結論

## 関連Issues, Discussions

#30

**Reply by kazzyfrog** [reply]
あと仕様の再確認もした方がいいかも！

FigJamや、Discussionsなど、色々に散らばっているかもなので！

**Reply by kazzyfrog** [reply]
## Discord、Issues、Discussionsの使い分けを確認
- Discord：カジュアルな連絡
- Issues：着手可能な、タスク、バグ報告
- Discussions：その他・作業記録、、

**Reply by kazzyfrog** [reply]
## イシューベースの開発の流れの確認

- イシュー立てる
- 自分を割り当てる
- 作業する（ブランチ切って、変更加える、コミット、push、PR）
- レビューに対応！

**Reply by kazzyfrog** [reply]
## ブランチ・コミットメッセージの命名規則

ドキュメントを参照！

**Reply by kazzyfrog** [reply]
## コードレビュー

- 作業記録を読みつつ、作業過程を追う。内容をざっくり把握して、レビューする！
- mainブランチのコードの品質は、低くてOK！後から修正すればOK！
- イシュー・PRにて、絶対満たすべきポイントと、それ以外のポイントで、分けて整理する！

**Reply by kazzyfrog** [reply]
一旦、自由にとっていくスタイルで！
- PRに関する、お知らせを、Discordに流す！
- 作業内容を、できるだけ、伝えるようにする！（PRの文章と、スレッドの作業記録）

**Comment by kagomen** [comment]
## トピック：デザインカンプについて
- デザインカンプ作成において、デザインの固定部分（使用カラーなど）のすり合わせ
  - 人気順・新着順で画像を表示する機能もすぐに実装しそう？なので、それに応じたデザインだと良いかも！
- 期限を設けたいので、都合の良い日を聞く

## 結論

## 関連Issues, Discussions

#47 
#50 
[FigJam](https://www.figma.com/board/bMwsq6U4qOEILPUk4E15NZ/LGTM-Factory?node-id=0-1&t=37gUC2FQu8miggpB-0)

**Reply by kazzyfrog** [reply]
構成として、近いサイトがあれば：
- [lucide](https://lucide.dev)
- [haikei](https://haikei.app)

**Reply by kazzyfrog** [reply]
## FigJamの確認

やったこと：
- デザインコンセプト決定
- サービス名も、現状変更なし
- ムードボード作成

**Reply by kagomen** [reply]
## ワイヤーフレームの作成
- します！
- 共有したのち、各々デザインカンプに着手！

**Comment by kagomen** [comment]
## トピック：これから（week3〜）の流れ
week3（8/19〜）からの流れの確認

- サイトのデザインカンプ作成
- LGTM画像のベースデザインを作成（各自好きなだけ）
- V1の機能実装
    - クリップボードにURLをコピーするボタンの機能
  - LGTM画像を生成するAPI

マイルストーンは、
ざっくり、こんな感じかな：
- W3/ デザインカンプと、機能実装
- →W4/ UI実装と、ドキュメントやサイトテキスト周り
- →確認作業で、V1ローンチ🚀

コツコツ、コンテンツ（LGTM画像のベースデザイン）を増やしていくのが、重要かも！

## 結論

## 関連Issues, Discussions

**Reply by kagomen** [reply]
以下、Discordからのコピー

```
webサイトのデザインができたら、
あとは、こんな感じが最低限かな
- LGTM画像のベースデザイン（コンテンツ）があって
- APIでの、動的なパラメータに応じた画像生成？できて
- 画像URLのコピーができる


あれば良い：
- クライアントサイドでの画像生成（プレビュー）→いらないかも。。
- ランダム生成
- 人気の画像＋内容の一覧？→こっちの方がいるかも

なので、先にできること：
- 画像生成まわりの機能（API、クライアントサイド）
- クリップボードにURLをコピーするボタンの機能
- LGTM画像のベースデザイン（コンテンツ）作成

```

**Reply by kagomen** [reply]
具体的なイシューとしては以下の感じになる？

### サイトのデザイン作成
- デザインカンプ作成→持ち寄り→最終的なカンプを決定

### LGTM画像のデザイン作成
- 各々ImageResponseを使って作る
  - src/app/api/ogディレクトリに、たくさんAPIを生やす感じ？
  - それをトップページで受け取って、一覧表示させる、という流れかな？
  - 何を動的な値にするかを決めたい
    - 個人的には、動的な値はテーマカラーやチーム名（または個人名）くらいでも良いのかも？、と思う
    - なぜなら、あまりにユーザー側のカスタムが自由だと、LGTM画像の数を増やせないから……
    - なので、カスタム機能の要件（？）としては、トップページで動的な値を操作して画像を一気に変化させる、のが良い気がする……→そうなるとクライアントサイドでの画像生成が必要になる……？👀

### クリップボードにURLをコピーするボタンの機能
- src/app/componentsディレクトリ

**Reply by kagomen** [reply]
@kazzyfrog 

LGTM画像の作成に関してまだよくイメージできていないのですが、上記の感じで合ってますか？
違っていたら、すり合わせしたいです……！！🙏

**Reply by kazzyfrog** [reply]
そのような認識で合ってると思います！👍

- たくさんAPIを生やす
- 何を動的な値にするか

ただ、上記に関しては、実装次第かなという感じです！
- １つのAPIで、JSX部分を動的に選択できる感じでも良いかも（type=simple 的な）
- たくさんAPI生やす場合は、動的な値は、それぞれでいいのかもなと思ってます！（テキスト、色、画像、GitHub URLなど、、）

この辺りは、先にベータ版っぽい実装をしてみないと、見えてこないかも、、と思っている感じです！

**Reply by kagomen** [reply]
> １つのAPIで、JSX部分を動的に選択できる感じでも良いかも（type=simple 的な）

なるほど！理解しました！🧠🙏🌟

> この辺りは、先にベータ版っぽい実装をしてみないと、見えてこないかも

たしかに……うーん……
イシュー化にあたって暫定的にでも決めたい感じがあるのですが、あんまりそういう進め方は良くないですかね……？

**Reply by kazzyfrog** [reply]
わかります！
その場合は、先にLGTM画像１つか２つ作って、
そのデザインに必要な、動的な値で進めてOKですよー！👍

自分、試作的に今日２つ作りましょうか？

**Reply by kagomen** [reply]
ちょっとあんまり実装のイメージができてないので、お時間があれば、お願いします！🙏

**Reply by kazzyfrog** [reply]
## V1の機能実装

- クリップボードにURLをコピーするボタンの機能
- LGTM画像を生成するAPI

**Reply by kazzyfrog** [reply]
一覧と、新着の表示機能は、一旦なしで！

**Comment by kazzyfrog** [comment]
## トピック：Week２関連の確認
焦点としては、こんな感じ
- 開発環境の整備
- デザインを詰める

やったこと：
- Next.jsセットアップ＆デプロイ
- テンプレート修正
- prettier設定
- FigJamで、
    - デザインコンセプト決定
    - サービス名も、現状変更なし
    - ムードボード作成

その他
- 画像生成機能の調査が開始している感じ

## 結論

## 関連Issues, Discussions

**Reply by kazzyfrog** [reply]
## 画像生成機能の調査
- #70 
- https://satori-image-response-practice.vercel.app/image-response
- https://playground.lgtm-factory.pages.dev/

**Reply by kazzyfrog** [reply]
## prettier設定

- `npm run format` を実行すれば、全体にフォーマットがかかる！
- VSCodeで設定していたら、セーブ時にフォーマッターが走る！
